src/WebJobs.Script/Rpc/FunctionRegistration/FunctionDispatcher.cs

```diff
@@ -107,12 +102,26 @@ internal async void InitializeJobhostLanguageWorkerChannelAsync()
             await InitializeJobhostLanguageWorkerChannelAsync(0);
         }
 
-        internal async Task InitializeJobhostLanguageWorkerChannelAsync(int attemptCount)
+        internal Task InitializeJobhostLanguageWorkerChannelAsync(int attemptCount)
         {
-            var languageWorkerChannel = _languageWorkerChannelFactory.CreateLanguageWorkerChannel(_scriptOptions.RootScriptPath, _workerRuntime, _metricsLogger, attemptCount, false, _managedDependencyOptions);
+            var languageWorkerChannel = _languageWorkerChannelFactory.CreateLanguageWorkerChannel(_scriptOptions.RootScriptPath, _workerRuntime, _metricsLogger, attemptCount, _managedDependencyOptions);
             languageWorkerChannel.SetupFunctionInvocationBuffers(_functions);
             _jobHostLanguageWorkerChannelManager.AddChannel(languageWorkerChannel);
-            await languageWorkerChannel.StartWorkerProcessAsync();
+            languageWorkerChannel.StartWorkerProcessAsync()
+                 .ContinueWith(workerInitTask =>
+                 {
+                     if (workerInitTask.IsCompleted)
+                     {
+                         _logger.LogDebug("Adding jobhost language worker channel for runtime: {language}. workerId:{id}", _workerRuntime, languageWorkerChannel.Id);
+                         languageWorkerChannel.SendFunctionLoadRequests();
+                         State = FunctionDispatcherState.Initialized;
+                     }
+                     else
+                     {
+                         _logger.LogWarning("Failed to start language worker process jobhost for runtime: {language}. workerId:{id}", _workerRuntime, languageWorkerChannel.Id);
+                     }
+                 });
+            return Task.CompletedTask;
```

Comment:
Not sure who the callers are here -- but this will not return immediately. Expected?

Reply:
Yes. This is intentional. Starting process and establishing grpc channel is a long running back ground task. We need the app to not block here. If we do receive invocation requests while the process is starting up, we use bufferblocks to buffer invocations and start executions once the process is ready.

Reply:
Does this need to return a Task? or could it be written as a sync method?

Reply:
This is intentionally async method so that it runs on a background  thread